hibernate-core/src/main/java/org/hibernate/engine/internal/TwoPhaseLoad.java

```diff
@@ -465,25 +428,14 @@ private static Boolean getOverridingEager(
 		// Performance: check type.isCollectionType() first, as type.isAssociationType() is megamorphic
 		if ( associationType.isCollectionType() || associationType.isAssociationType()  ) {
 
-			// check 'fetch graph' first; skip 'fetch profile' if 'fetch graph' takes effect
-			Boolean overridingEager = isEagerFetchGraph( session, associationName, associationType );
-
-			if ( overridingEager != null ) {
-				//This method is very hot, and private so let's piggy back on the fact that the caller already knows the debugging state.
-				if ( isDebugEnabled ) {
-					LOG.debugf(
-							"Overriding eager fetching using fetch graph. EntityName: %s, associationName: %s, eager fetching: %s",
-							entityName,
-							associationName,
-							overridingEager
-					);
-				}
-
-				return overridingEager;
+			// we can return false invariably for if the entity has been covered by entity graph,
+			// its associated JOIN has been present in the SQL generated and hence it would be loaded anyway
+			if ( session.isEnforcingFetchGraph() ) {
+				return false;
```

Comment:
The above one-liner is the gist of the ultimate fix. Its comment is self-exlanatory.